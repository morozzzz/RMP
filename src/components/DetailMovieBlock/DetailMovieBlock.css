.detail-block {
    width: 100%;
    background: url('../assets/background_main.jpg');
    background-size: cover;
}

.detail-block__overlay {
    background-color: rgba(0, 0, 0, 0.781);
    height: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
}

.detail-block__top-bar {
    width: 80%;
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 0.8em 0;
}

.detail-block__search-button {
    outline: none;
    background: rgb(66, 66, 66);
    border: none;
    color: rgb(248, 129, 129);
    font-weight: bold;
    height: 2em;
    width: 5em;
    cursor: pointer;    
}

.detail-block__content {
    max-width: 1080px;
    width: 40%;
    display: flex;
    justify-content: space-between;
    margin: 1em auto;
}

.detail-block__image-container {
    width: 35%;    
}

.detail-block__image {
    width: 100%;
}

.detail-block__info {
    width: 55%;
    color: white;
}

.detail-block__primary-data {
    display: flex;
    align-items: center;
    margin: 0.4em 0;
}

.detail-block__title {
    font-size: 2em;
    color: rgb(248, 129, 129);
}

.detail-block__additional {
    margin: 0.2em 0;
}

.detail-block__rating {
    font-size: 1.3em;
    border: 0.05em solid rgb(185, 250, 132);
    padding: 0.2em;
    border-radius: 50%;
    color: rgb(185, 250, 132);
    margin-left: 1em;
}

.detail-block__secondary-data {
    margin: 0.8em 0;
    display: flex;
    width: 28%;
    justify-content: space-between;
    font-weight: bold;   
}

.detail-block__description {
    margin: 1em 0;
    line-height: 1.25em;
}
