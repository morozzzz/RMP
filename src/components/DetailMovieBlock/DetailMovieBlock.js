// @flow

import React from 'react';
import withStyles from 'react-jss';
import Logo from '../Logo/Logo';

type detailMovieBlockProps = {
    movieData: Object,
    onSearchClick: Function
};

const backgroundImgUrl = '../../assets/background_main.jpg';

const styles = {
    // .detail-block
    detailBlockClass: {
        width: '100%',
        background: `url(${backgroundImgUrl})`,
        backgroundSize: 'cover',
    },
    // .detail-block__overlay
    detailBlockOverlayClass: {
        backgroundColor: 'rgba(0, 0, 0, 0.781)',
        height: '100%',
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
    },
    // .detail-block__top-bar
    detailBlockTopBarClass: {
        width: '80%',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'space-between',
        padding: '0.8em 0',
    },
    // .detail-block__search-button
    detailBlockSearchButtonClass: {
        outline: 'none',
        background: 'rgb(66, 66, 66)',
        border: 'none',
        color: 'rgb(248, 129, 129)',
        fontWeight: 'bold',
        height: '2em',
        width: '5em',
        cursor: 'pointer',
    },
    // .detail-block__content
    detailBlockContentClass: {
        maxWidth: '1080px',
        width: '40%',
        display: 'flex',
        justifyContent: 'space-between',
        margin: '1em auto',
    },
    // .detail-block__image-container
    detailBlockImageContainerClass: {
        width: '35%',
    },
    // .detail-block__image
    detailBlockImageClass: {
        width: '100%',
    },
    // .detail-block__info
    detailBlockInfoClass: {
        width: '55%',
        color: 'white',
    },
    // .detail-block__primary-data
    detailBlockPrimaryDataClass: {
        display: 'flex',
        alignItems: 'center',
        margin: '0.4em 0',
    },
    // .detail-block__title
    detailBlockTitleClass: {
        fontSize: '2em',
        color: 'rgb(248, 129, 129)',
    },
    // .detail-block__additional
    detailBlockAdditionalClass: {
        margin: '0.2em 0',
    },
    // .detail-block__rating
    detailBlockRatingClass: {
        fontSize: '1.3em',
        border: '0.05em solid rgb(185, 250, 132)',
        padding: '0.2em',
        borderRadius: '50%',
        color: 'rgb(185, 250, 132)',
        marginLeft: '1em',
        minWidth: '1.2em',
        minHeight: '1.2em',
        textAlign: 'center',
    },
    // .detail-block__secondary-data
    detailBlockSecondaryDataClass: {
        margin: '0.8em 0',
        display: 'flex',
        width: '100%',
        justifyContent: 'space-between',
        fontWeight: 'bold',
    },
    // .detail-block__description
    detailBlockDescriptionClass: {
        margin: '1em 0',
        lineHeight: '1.25em',
    },
};

const DetailMovieBlock = (props: detailMovieBlockProps) => {
    const {
        movieData: {
            title,
            release_date,
            runtime,
            vote_average,
            tagline,
            poster_path,
            overview,
        },
        onSearchClick,
        classes,
    } = props;

    return (
        <div className={classes.detailBlockClass}>
            <div className={classes.detailBlockOverlayClass}>
                <div className={classes.detailBlockTopBarClass}>
                    <Logo name="netflixroulettee" redirectUrl="" />
                    <button onClick={onSearchClick} type="button" className={classes.detailBlockSearchButtonClass}>SEARCH</button>
                </div>
                {
                    title ? (
                        <div className={classes.detailBlockContentClass}>
                            <div className={classes.detailBlockImageContainerClass}>
                                <img className={classes.detailBlockImageClass} src={poster_path} alt="poster" />
                            </div>
                            <div className={classes.detailBlockInfoClass}>
                                <div className={classes.detailBlockPrimaryDataClass}>
                                    <div className={classes.detailBlockTitleClass}>{title}</div>
                                    <div className={classes.detailBlockRatingClass}>
                                        {vote_average}
                                    </div>
                                </div>
                                <div className={classes.detailBlockAdditionalClass}>{tagline}</div>
                                <div className={classes.detailBlockSecondaryDataClass}>
                                    <div>{release_date}</div>
                                    <div>{`${runtime} min`}</div>
                                </div>
                                <div className={classes.detailBlockDescriptionClass}>
                                    {overview}
                                </div>
                            </div>
                        </div>
                    )
                        : null
                }
            </div>
        </div>
    );
};

export default withStyles(styles)(DetailMovieBlock);
