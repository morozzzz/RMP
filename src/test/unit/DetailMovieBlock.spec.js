import React from 'react';
import { shallow } from 'enzyme';
import DetailMovieBlock from '../../components/DetailMovieBlock/DetailMovieBlock';

jest.mock('../../components/Logo/Logo', () => () => <div />);

const data = {
    title: 'Some title',
    release_date: '2002',
    runtime: '156',
    vote_average: '10',
    tagline: 'Some tagline',
    poster_path: 'http://some_url',
    overview: 'Some overview',
};

describe('DetailMovieBlock', () => {
    it('should be rendered correctly', () => {
        const component = shallow(<DetailMovieBlock movieData={data} />);
        expect(component).toMatchSnapshot();
    });

    it('click on SEARCH button should call props.onSearchClick', () => {
        const onSearchClickMock = jest.fn();
        const component = shallow(
            <DetailMovieBlock movieData={data} onSearchClick={onSearchClickMock} />,
        );
        const serachButton = component.find('.detail-block__search-button');

        serachButton.simulate('click');

        expect(onSearchClickMock).toHaveBeenCalledTimes(1);
    });
});
