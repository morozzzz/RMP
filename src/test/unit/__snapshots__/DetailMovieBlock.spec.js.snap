// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailMovieBlock should be rendered correctly 1`] = `
<div
  className="detail-block"
>
  <div
    className="detail-block__overlay"
  >
    <div
      className="detail-block__top-bar"
    >
      <Component
        name="netflixroulettee"
        redirectUrl=""
      />
      <button
        className="detail-block__search-button"
        type="button"
      >
        SEARCH
      </button>
    </div>
    <div
      className="detail-block__content"
    >
      <div
        className="detail-block__image-container"
      >
        <img
          alt="poster"
          className="detail-block__image"
          src="http://some_url"
        />
      </div>
      <div
        className="detail-block__info"
      >
        <div
          className="detail-block__primary-data"
        >
          <div
            className="detail-block__title"
          >
            Some title
          </div>
          <div
            className="detail-block__rating"
          >
            10
          </div>
        </div>
        <div
          className="detail-block__additional"
        >
          Some tagline
        </div>
        <div
          className="detail-block__secondary-data detail-block-secondary-data"
        >
          <div
            className="detail-block-secondary-data__item"
          >
            2002
          </div>
          <div
            className="detail-block-secondary-data__item"
          >
            156 min
          </div>
        </div>
        <div
          className="detail-block__description"
        >
          Some overview
        </div>
      </div>
    </div>
  </div>
</div>
`;
